server/transport.go

```diff
@@ -104,11 +109,21 @@ func message(r *http.Request) ([]byte, error) {
 		}
 		op := q.Get("operation")
 		if op == "PKIOperation" {
+			if len(msg) == 0 {
+				return nil, &BadRequestError{Message: "missing PKIOperation message"}
+			}
```

Comment:
Here too, since we're adding new validation, just to make sure that we're not erroring on what is possibly a valid request that was previously ignored?

Reply:
agreed on this one as well.

Reply:
I'm thinking: what about clients that use `POST`? (in SCEP you use a single path, and can use `GET` or `POST`, all communications are handled by parameters)

wonder if we should add the check upstream so it works independent of the request method used?

https://github.com/fleetdm/scep/blob/3aa7a5a9d937a028c081e46f605c4a85fbcdadde/server/service.go#L75

Reply:
@roperzh oh, I didn't realize that was the case! yeah, if you can use POST or GET for PKIOperations, then we should check before here. I can update that.

Reply:
@jahzielv I think that won't do it for POST because the message comes in the request body 🤔 

If it's just `operation=PKIOperation` we're worried about (which seems reasonable for now), what do you think about doing it in the service (link I pointed above)

```
if len(data) == 0 {
    // bad request!
}
```

Reply:
🤦 gotcha. yeah, needed a small update to another part of the code, but that works! updated @roperzh